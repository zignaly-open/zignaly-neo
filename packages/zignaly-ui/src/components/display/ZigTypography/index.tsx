import { Typography as ZigTypography } from "@mui/material";

export default ZigTypography;
