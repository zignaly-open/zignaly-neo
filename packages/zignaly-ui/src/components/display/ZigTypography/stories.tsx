import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Typography } from "@mui/material";
import { Variant } from "@mui/material/styles/createTypography";

export default {
  title: "Display/ZigTypography",
  component: Typography,
  argTypes: {
    text: {
      control: "text",
      defaultValue: "Hello",
    },
    variant: {
      options: ["body1", "body2", "bigNumber", "h1", "h2", "h3", "h4", "caption"],
      control: { type: "select" },
      defaultValue: "body1",
    },
  },
  parameters: {
    jest: ["typography"],
  },
} as ComponentMeta<typeof Typography>;

const Template: ComponentStory<React.FC<{ text: string; variant: string }>> = ({
  variant,
  text,
}) => (
  <Typography component={"p"} variant={variant as Variant}>
    {text}
  </Typography>
);

export const Basic = Template.bind({});
Basic.args = {
  text: "Example text",
};
