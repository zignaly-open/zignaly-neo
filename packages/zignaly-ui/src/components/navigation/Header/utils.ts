// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
