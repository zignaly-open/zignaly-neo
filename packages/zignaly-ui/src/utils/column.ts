import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
