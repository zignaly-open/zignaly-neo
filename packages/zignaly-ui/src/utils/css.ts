import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
