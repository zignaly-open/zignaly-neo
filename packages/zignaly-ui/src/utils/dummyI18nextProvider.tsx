import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
