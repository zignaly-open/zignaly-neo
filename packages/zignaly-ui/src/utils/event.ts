export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
