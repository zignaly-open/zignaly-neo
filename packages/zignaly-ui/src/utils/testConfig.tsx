import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
