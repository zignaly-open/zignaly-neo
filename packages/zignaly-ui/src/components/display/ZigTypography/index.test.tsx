import React from "react";
import ZigTypography from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/display/ZigTypography", () => {
  describe("renders", () => {
    it("ZigTypography render", async () => {
      const { container } = renderWithProvidersUi(
        <ZigTypography id={"typography"}></ZigTypography>,
      );

      const typography = container.querySelector("#typography") as Element;
      expect(typography).toBeVisible();
      const typographyStyles = getComputedStyle(typography);
      expect(typography).toMatchSnapshot();
      expect(typographyStyles).toMatchSnapshot();
    });
    it("ZigTypography render custom variant", async () => {
      const { container } = renderWithProvidersUi(
        <ZigTypography id={"typography"} variant={"h2"}></ZigTypography>,
      );

      const typography = container.querySelector("#typography") as Element;
      expect(typography).toBeVisible();
      const typographyStyles = getComputedStyle(typography);
      expect(typography).toMatchSnapshot();
      expect(typographyStyles).toMatchSnapshot();
    });
    it("ZigTypography render with sx prop", async () => {
      const { container } = renderWithProvidersUi(
        <ZigTypography id={"typography"} sx={{ mt: "15px", fontSize: "25px" }}></ZigTypography>,
      );

      const typography = container.querySelector("#typography") as Element;
      expect(typography).toBeVisible();
      const typographyStyles = getComputedStyle(typography);
      expect(typography).toMatchSnapshot();
      expect(typographyStyles).toMatchSnapshot();
    });
  });
});
