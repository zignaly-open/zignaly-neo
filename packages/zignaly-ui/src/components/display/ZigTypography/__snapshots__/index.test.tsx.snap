// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigTypography renders with custom variant 1`] = `
<h2
  class="MuiTypography-root MuiTypography-h2 css-1hohiwo-MuiTypography-root"
  id="typography"
/>
`;

exports[`components/display/ZigTypography renders with custom variant 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "font-size",
  "2": "font-weight",
  "3": "margin",
  "4": "font-family",
  "5": "line-height",
  "6": "letter-spacing",
  "7": "color",
  "8": "visibility",
  "_importants": {
    "color": "",
    "display": "",
    "font-family": "",
    "font-size": "",
    "font-weight": "",
    "letter-spacing": "",
    "line-height": "",
    "margin": "",
    "visibility": undefined,
  },
  "_length": 9,
  "_onChange": [Function],
  "_values": {
    "color": "rgb(193, 193, 200)",
    "display": "block",
    "font-family": "Avenir Next,Roboto,Helvetica,Arial,sans-serif",
    "font-size": "18px",
    "font-weight": "500",
    "letter-spacing": "0.55px",
    "line-height": "28px",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigTypography renders with default props 1`] = `
<span
  class="MuiTypography-root MuiTypography-body1 css-15f8135-MuiTypography-root"
  id="typography"
/>
`;

exports[`components/display/ZigTypography renders with default props 2`] = `
CSSStyleDeclaration {
  "0": "margin",
  "1": "font-family",
  "2": "font-weight",
  "3": "font-size",
  "4": "line-height",
  "5": "letter-spacing",
  "6": "color",
  "7": "visibility",
  "_importants": {
    "color": "",
    "font-family": "",
    "font-size": "",
    "font-weight": "",
    "letter-spacing": "",
    "line-height": "",
    "margin": "",
    "visibility": undefined,
  },
  "_length": 8,
  "_onChange": [Function],
  "_values": {
    "color": "rgb(169, 169, 186)",
    "font-family": "Avenir Next,Roboto,Helvetica,Arial,sans-serif",
    "font-size": "15px",
    "font-weight": "400",
    "letter-spacing": "0.55px",
    "line-height": "24px",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "visibility": "visible",
  },
}
`;

exports[`components/display/ZigTypography renders with sx prop 1`] = `
<span
  class="MuiTypography-root MuiTypography-body1 css-f7uwmi-MuiTypography-root"
  id="typography"
/>
`;

exports[`components/display/ZigTypography renders with sx prop 2`] = `
CSSStyleDeclaration {
  "0": "margin",
  "1": "font-family",
  "2": "font-weight",
  "3": "font-size",
  "4": "line-height",
  "5": "letter-spacing",
  "6": "color",
  "7": "visibility",
  "_importants": {
    "color": "",
    "font-family": "",
    "font-size": "",
    "font-weight": "",
    "letter-spacing": "",
    "line-height": "",
    "margin": undefined,
    "margin-top": "",
    "visibility": undefined,
  },
  "_length": 8,
  "_onChange": [Function],
  "_values": {
    "color": "rgb(169, 169, 186)",
    "font-family": "Avenir Next,Roboto,Helvetica,Arial,sans-serif",
    "font-size": "25px",
    "font-weight": "400",
    "letter-spacing": "0.55px",
    "line-height": "24px",
    "margin": "15px 0px 0px 0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "15px",
    "visibility": "visible",
  },
}
`;
